monitor/payload/monitor_payload.go

```diff
@@ -105,6 +105,11 @@ func ReadMetaPayload(r io.Reader, meta *Meta, pl *Payload) error {
 		return err
 	}
 
+	// If there is no payload just abort
+	if meta.Size == 0 {
+		return nil
```

Comment:
The implied behaviour here is that payload's contents are undefined in the `meta.Size==0` case. We can zero out payload or add a comment about this in the function comment (line 102)